### PR TITLE
Telemetry: Use interval from server

### LINKED ISSUE #529: S3GW Telemetry: Use requestIntervalInMinutes instead of rgw_s3gw_telemetry_update_interval config
Use the requestIntervalInMinutes value from the upgrade responder response instead of the rgw_s3gw_telemetry_update_interval RGW configuration. Needs changes to the on/off logic and a new configuration option for that.

Follow up to #423

### PR DESCRIPTION
**Pull request by irq0**
Record requestIntervalInMinutes sent back from the upgrade responder
to set the retry interval in the future. Initialize interval with 10
minutes, overwritten after the first successful response.

Signed-off-by: Marcel Lauhoff <marcel.lauhoff@suse.com>Fixes https://github.com/aquarist-labs/s3gw/issues/529

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

### CODE REVIEW

File: src/rgw/rgw_s3gw_telemetry.cc [hunk 1]
```diff
@@ -199,23 +193,50 @@ void S3GWTelemetry::Version::decode_json(JSONObj* obj) {
   JSONDecoder::decode_json("ReleaseDate", release_date, obj, true);
 }
 
-std::optional<std::vector<S3GWTelemetry::Version>>
-S3GWTelemetry::parse_upgrade_response(bufferlist& response) const {
+bool S3GWTelemetry::parse_upgrade_response(
+    bufferlist& response, std::chrono::milliseconds* out_update_interval,
+    std::vector<Version>* out_versions
+) const {
   std::vector<Version> result;
```
Comment (0xavi0) [inline src/rgw/rgw_s3gw_telemetry.cc]: `result` is no longer used.

File: src/rgw/rgw_s3gw_telemetry.cc [hunk 2]
```diff
@@ -199,23 +193,50 @@ void S3GWTelemetry::Version::decode_json(JSONObj* obj) {
   JSONDecoder::decode_json("ReleaseDate", release_date, obj, true);
 }
 
-std::optional<std::vector<S3GWTelemetry::Version>>
-S3GWTelemetry::parse_upgrade_response(bufferlist& response) const {
+bool S3GWTelemetry::parse_upgrade_response(
+    bufferlist& response, std::chrono::milliseconds* out_update_interval,
+    std::vector<Version>* out_versions
```
Comment (0xavi0) [inline src/rgw/rgw_s3gw_telemetry.cc]: Why using pointers for those 2 return values? 
Code is not checking if passed value is `nullptr` before using them, so I guess we are always going to set their values.

I suggest using references for them.

File: src/rgw/rgw_s3gw_telemetry.cc (hunk 1 above)
Comment (irq0) [reply src/rgw/rgw_s3gw_telemetry.cc]: Thanks, good catch!

**Review by 0xavi0 — DISMISSED** [review]
lgtm, looks like you need to fix the tests.

**Review by 0xavi0 — APPROVED** [review]

**Pull request merged**